src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -4269,10 +4299,10 @@ internal int VisibleCount
         /// <returns>the enumerator for this collection</returns>
         public override IEnumerator<T> GetEnumerator()
         {
-            lock (_members)
+            lock (Members)
```

Comment:
if we want to go bit far, maybe also check `_members == null` here and if so return `Array.Empty<T>.GetEnumerator()`?
Just an idea.

Reply:
Good idea for the _public_ method.

Fixed as `return Enumerable.Empty<T>().GetEnumerator();`